app/models/attributes/policies_config.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class Attributes::PoliciesConfig < ActiveRecord::Type::Text
+  def cast(value)
+    if value.is_a?(Proxy::PoliciesConfig)
```

Comment:
Code Climate will complain for sure :)

Anyway I do not think we need this, because `#cast` is called when retrieving values from the database.

Reply:
I can't remember code climate complaining. But I can't review weaved warnings. Trying to filter them out yields nothing.

This casting was actually needed, IIRC for the `*changed?`methods.

Reply:
I ran locally to double check:
```
$ reek app/models/attributes/policies_config.rb
Inspecting 1 file(s):
S

app/models/attributes/policies_config.rb -- 2 warnings:
  [4]:UtilityFunction: Attributes::PoliciesConfig#cast doesn't depend on instance state (maybe move it to another class?) [https://github.com/troessner/reek/blob/v6.0.4/docs/Utility-Function.md]
  [12]:UtilityFunction: Attributes::PoliciesConfig#serialize doesn't depend on instance state (maybe move it to another class?) [https://github.com/troessner/reek/blob/v6.0.4/docs/Utility-Function.md]
```

And using instance methods is how custom types are supposed to be defined as far as I can see. So the 2 errors are ignored. There is no complaint about type checking. And it was needed, that's why I added. Do you want me to investigate where it was needed?

Reply:
Well it should :) https://github.com/troessner/reek/blob/master/docs/Simulated-Polymorphism.md
Maybe we allow it in our configuration :(

Reply:
Seems like it is not needed. Maybe it failed before some fixes. As I wrote to the other comment, `#deserialize` just delegates to `#cast` so they are almost the same thing.